FWCore/Framework/test/test_catch2notTP_InputProcessBlockCacheHolder.cc

```diff
@@ -0,0 +1,168 @@
+    REQUIRE(edm::impl::countTypeInParameterPack<edmtest::TestProcessBlockCacheA,
+                                                edmtest::TestProcessBlockCacheA,
+                                                edmtest::TestProcessBlockCacheA,
+                                                edmtest::TestProcessBlockCacheC>() == 2);
+    REQUIRE(edm::impl::countTypeInParameterPack<edmtest::TestProcessBlockCacheA,
+                                                edmtest::TestProcessBlockCacheB,
+                                                edmtest::TestProcessBlockCacheB,
+                                                edmtest::TestProcessBlockCacheA>() == 1);
+    REQUIRE(edm::impl::countTypeInParameterPack<edmtest::TestProcessBlockCacheA,
+                                                edmtest::TestProcessBlockCacheB,
+                                                edmtest::TestProcessBlockCacheB,
+                                                edmtest::TestProcessBlockCacheC>() == 0);
+    REQUIRE(edm::impl::countTypeInParameterPack<edmtest::TestProcessBlockCacheA,
+                                                edmtest::TestProcessBlockCacheB,
+                                                edmtest::TestProcessBlockCacheA,
+                                                edmtest::TestProcessBlockCacheA>() == 2);
+    REQUIRE(edm::impl::countTypeInParameterPack<edmtest::TestProcessBlockCacheA,
+                                                edmtest::TestProcessBlockCacheB,
+                                                edmtest::TestProcessBlockCacheA,
+                                                edmtest::TestProcessBlockCacheC>() == 1);
+
+    // The following should not compile and I manually verified it doesn't
+    // REQUIRE(edm::impl::countTypeInParameterPack<>() == 0);
+  }
+
+  SECTION("test indexInputProcessBlockCache") {
+    REQUIRE(
+        edm::impl::indexInputProcessBlockCache<edmtest::TestProcessBlockCacheA, edmtest::TestProcessBlockCacheA>() ==
+        0);
+    REQUIRE(edm::impl::indexInputProcessBlockCache<edmtest::TestProcessBlockCacheA,
+                                                   edmtest::TestProcessBlockCacheA,
+                                                   edmtest::TestProcessBlockCacheB>() == 0);
+    REQUIRE(edm::impl::indexInputProcessBlockCache<edmtest::TestProcessBlockCacheA,
+                                                   edmtest::TestProcessBlockCacheB,
+                                                   edmtest::TestProcessBlockCacheA>() == 1);
+    REQUIRE(edm::impl::indexInputProcessBlockCache<edmtest::TestProcessBlockCacheD,
+                                                   edmtest::TestProcessBlockCacheA,
+                                                   edmtest::TestProcessBlockCacheB,
+                                                   edmtest::TestProcessBlockCacheC,
+                                                   edmtest::TestProcessBlockCacheD>() == 3);
```

Comment:
Tests in this file up to here are also compile-time checks, right?

Reply:
Done. I changed them to use STATIC_REQUIRE. That should be better.